### ISSUE TITLE
te_trc_update tool is broken

### ISSUE THREAD

**Issue by okt-denispr** [oktetlabs/test-environment#17]
Hi, I found that `te_trc_update_wrapper` tool is broken.
When I'm trying to use it I see:
```
te_tester: ../src/engine/tester/test_path.c:991: merge_test_paths: Assertion `FALSE' failed.
```
This problem is caused by the following commit:
```
commit 2bd667e9e6acd25b60d771a61a46f939f50b03fe
Author: Ivan Khodyrev <ivan.khodyrev@oktetlabs.ru>
Date:   Tue Jun 20 14:19:50 2023 +0300

    engine/tester: add opt for tester to fail on memory leaks
    
    At the moment, it is possible to track memory leaks for a
    specific test only by viewing the valgrind log files for
    each test, which is quite time-consuming. The patch adds a
    flag that causes the tests in which a memory leak occurs to
    fail.
    
    OL-Redmine-Id: 12963
    Signed-off-by: Ivan Khodyrev <ivan.khodyrev@oktetlabs.ru>
    Reviewed-by: Artem Andreev <artem.andreev@oktetlabs.ru>
```

`TESTER_OPT_FAIL_ON_LEAK` breaks the order for values that must correspond to `test_path_typeof` in `process_cmd_line_opts` function.

Could you please fix it?

**Comment by ol-khodyrev** [comment]
@okt-denispr, please write an example of how to reproduce this bug.

**Comment by ol-khodyrev** [comment]
Yes, I have already fixed this error, I just want to check that everything is working correctly now.

**Comment by ol-arteman** [comment]
Fixed in release v1.16.0 (705d9ae3caf9dc82155e74b7af12b0bd48f106dc)
